torch/serialization.py

```diff
@@ -453,6 +453,12 @@ def persistent_load(saved_id):
             raise RuntimeError("Unknown saved id type: %s" % saved_id[0])
 
     f_is_real_file = _is_real_file(f)
+
+    if not hasattr(f, 'seek'):
+        raise RuntimeError("You can only torch.load from a file that is seekable. \
+        Please read the file into a seekable file on disk using request.urlretrieve \
```

Comment:
Well `request.urlretrieve` is a hint that's pretty specific to a certain use case. I'd rather say that you should pre-load the data into a buffer like `io.BytesIO` and load from that.